config/config.go

```diff
@@ -399,8 +394,33 @@ func (c KubeClusterSettings) K8sNetworkPlugin() string {
 func (c Cluster) Config() (*Config, error) {
 	config := Config{Cluster: c}
 
-	config.MinControllerCount = config.ControllerCount - 1
-	config.MaxControllerCount = config.ControllerCount + 1
+	// setup controller ASG size, one is the default ControllerCount
+	if config.ControllerCount != 1 && (config.ControllerASG.MinSize != 0 || config.ControllerASG.MaxSize != 0) {
+		return nil, fmt.Errorf("`controllerASG.MinSize` and `controllerASG.MaxSize` can only be specified without `controllerCount`")
+	}
+	if config.ControllerASG.MinSize == 0 {
+		config.ControllerASG.MinSize = config.ControllerCount
+	}
+	if config.ControllerASG.MaxSize == 0 {
+		config.ControllerASG.MaxSize = util.IntMax(config.ControllerCount, config.ControllerASG.MinSize)
+	}
+	if config.ControllerASG.RollingUpdateMinInstancesInService == 0 {
+		config.ControllerASG.RollingUpdateMinInstancesInService = config.ControllerASG.MaxSize - 1
+	}
+
+	// setup worker ASG size, one is the default WorkerCount
+	if config.WorkerCount != 1 && (config.WorkerASG.MinSize != 0 || config.WorkerASG.MaxSize != 0) {
+		return nil, fmt.Errorf("`workerASG.MinSize` and `workerASG.MaxSize` can only be specified without `workerCount`")
+	}
+	if config.WorkerASG.MinSize == 0 {
+		config.WorkerASG.MinSize = config.WorkerCount
+	}
```

Comment:
Could we just alter `func (c WorkerSettings) MinWorkerCount() int` to achieve the same end result?

e.g.

```
func (c WorkerSettings) MinWorkerCount() int {
	if c.WorkerASG.MinSize == 0 {
		return c.WorkerCount
	}
	return c.WorkerASG.MinSize
	// Instead of:
	// return c.WorkerCount - 1
}
```

So that we don't even need to modify `stack-template.json` and could keep backward-compatibility of what are (exposed to|usable from) stack-template.json